scripts/update_and_rebuild_libmesh.sh

```diff
@@ -5,24 +5,50 @@ DIAGNOSTIC_LOG="libmesh_diagnostic.log"
 # Set go_fast flag if "--fast" is found in command line args.
 for i in "$@"
 do
+  shift
   if [ "$i" == "--fast" ]; then
     go_fast=1;
-    break;
+  fi
+
+  if [[ "$i" == "-h" || "$i" == "--help" ]]; then
+    help=1;
+  fi
+
+  if [ "$i" == "--skip-submodule-update" ]; then
+    skip_sub_update=1;
+  else # Remove the skip submodule update argument before passing to libMesh configure
+    set -- "$@" "$i"
   fi
 done
 
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Display help
+if [[ -n "$help" ]]; then
+  cd $SCRIPT_DIR/..
+  echo "Usage: $0 [-h | --help | --fast | --skip-submodule-update | <libmesh options> ]"
+  echo
+  echo "-h | --help              Display this message and list of available libmesh options"
+  echo "--fast                   Run libmesh 'make && make install' only, do NOT run configure"
+  echo "--skip-submodule-update  Do not update the libMesh submodule, use the current version"
+  echo "*************************************************************************************"
+  echo ""
+
+  libmesh/configure -h
```

Comment:
If the user has never initialized their libmesh submodule, this won't work...

Reply:
Yeah, I thought about that. I guess I can protect it with another if statement.